overlord/ifacestate/ifacemgr.go

```diff
@@ -275,9 +275,23 @@ func (m *InterfaceManager) doRemoveSnapSecurity(task *state.Task, _ *tomb.Tomb)
 	if err != nil {
 		return err
 	}
-	// TODO: remove all connections from the state
-	if err := m.repo.RemoveSnap(snapName); err != nil {
 
+	conns, err := getConns(task.State())
```

Comment:
This makes a lot of sense. Earlier when I was working on ``setup-snap-security`` I realized that it would be much easier if there were some more fine-grained tasks (e.g. setup that could be reused by connect) and it looks like the same is true for ``remove-snap-security``.

So let's review how each of those would be used:

``setup-profiles``: on ``snap install``, ``snap refresh`` and maybe ``snap activate``
``remove-profiles``: on ``snap deactivate``
``discard-conns``: on ``snap remove``

@niemeyer @mvo5 ^^ does that make sense?